### PR TITLE
operator/v1: Make .spec.replicas required

### PR DESCRIPTION
**Pull request by Miciah**
This field is required for the `/scale` subresource.

* `operator/v1/types_ingress.go` (`IngressControllerSpec`): Make `replicas` required.

### CODE REVIEW

**Comment by ironcladlou** [comment]
Need to run `make generate`

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -55,11 +55,8 @@ type IngressControllerSpec struct {
 	// +optional
 	Domain string `json:"domain,omitempty"`
 
-	// replicas is the desired number of ingress controller replicas. If unset,
-	// defaults to 2.
-	//
-	// +optional
-	Replicas *int32 `json:"replicas,omitempty"`
+	// replicas is the desired number of ingress controller replicas.
+	Replicas int32 `json:"replicas"`
```
Comment (ironcladlou) [inline operator/v1/types_ingress.go]: @smarterclayton is there a reason you wanted this to be a pointer? Seems like it has to be a value type for scale subresource to work.

Comment (smarterclayton) [reply operator/v1/types_ingress.go]: Every workload object should have pointer to replicas - see RS, STS, Deployments, etc.

Comment (smarterclayton) [reply operator/v1/types_ingress.go]: Why are you different than deployments which have a scale resource?  If you say "scale X" of course you have specify X.

Comment (ironcladlou) [reply operator/v1/types_ingress.go]: I see that now... must something we're missing wrt scale

**Comment by ironcladlou** [comment]
@Miciah I think we were wrong about this one and want to keep the pointer

**Comment by Miciah** [comment]
Sorry! Regenerated.

**Comment by ironcladlou** [comment]
Given our CRD definition:

```
subresources:
    status: {}
    scale:
      specReplicasPath: .spec.replicas
      statusReplicasPath: .status.availableReplicas
      labelSelectorPath: .status.labelSelector
```

When using the pointer form:

```
$ oc scale --replicas=1 -n openshift-ingress-operator ingresscontrollers/default                                                                                                                               
Error from server (InternalError): Internal error occurred: the spec replicas field ".spec.replicas" does not exist
```

Is this a bug in upstream CRD handling?

**Comment by ironcladlou** [comment]
cc @deads2k @openshift/sig-master

**Comment by ironcladlou** [comment]
Talking with @soltysh and @sttts this could actually be a bug in k8s scale implementation. The current pointer API is consistent with desired conventions, and we can make a note about `oc scale` being problematic and also work on a fix, so sounds like we don't need this change/PR?

**Comment by Miciah** [comment]
You beat me to it.  Yup, agreed.  I'll close this PR.

**Pull request closed without merging**